src/mots/bmo.py

```diff
@@ -0,0 +1,108 @@
+"""Module that provides helpers to interact with the Bugzilla API."""
+
+# TODO: this should be abstracted so that it can be hot-swappable with any API that can
+# be searched.
+
+from __future__ import annotations
+import requests
+import os
+import logging
+import time
+
+
+logger = logging.getLogger(__name__)
+
+DEFAULT_BASE_URL = "https://bugzilla.mozilla.org/rest"
+USER_AGENT = "mots"  # TODO: improve this and include version.
+
+# TODO set NO_INPUT var to prevent prompt.
+
+
+class BMOClient:
+    """A thin wrapper as a Bugzilla API client."""
+
+    def __init__(self, token: str = None, base_url: str = DEFAULT_BASE_URL):
+        if not token:
+            token = os.getenv("BUGZILLA_API_KEY", "")
+            if not token:
+                token = input("Enter BMO Token: ")
+            if not token:
+                raise Exception()
+        self.headers = {"X-BUGZILLA-API-KEY": token, "User-Agent": USER_AGENT}
+        self.base_url = base_url
+
+    def _get(self, path: str, params=None):
+        logger.debug(f"GET {self.base_url}/{path}")
+        params = params or {}
+        response = requests.get(
+            f"{self.base_url}/{path}", headers=self.headers, params=params
+        )
+        return response
+
+    def get_user(self, email: str, rate_limit_delay: float = 0.3):
+        """Get user data based on provided email address."""
+        time.sleep(rate_limit_delay)
```

Comment:
I don't think sleeping before making any requests to BMO is the right approach. We could record the timestamp of the request as `self.last_request` and wait if two requests are within the rate limit delay instead. What do you think?

Reply:
This is actually obsolete now, since everything is done in a single API request, I will update this code and remove the rate limits.